src/test/SkipCallsTestHook.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import {Hooks} from "../libraries/Hooks.sol";
+import {BaseTestHooks} from "./BaseTestHooks.sol";
+import {IHooks} from "../interfaces/IHooks.sol";
+import {IPoolManager} from "../interfaces/IPoolManager.sol";
+import {PoolKey} from "../types/PoolKey.sol";
+import {BalanceDelta} from "../types/BalanceDelta.sol";
+import {PoolId, PoolIdLibrary} from "../types/PoolId.sol";
+import {IERC20Minimal} from "../interfaces/external/IERC20Minimal.sol";
+import {CurrencyLibrary, Currency} from "../types/Currency.sol";
+import {PoolTestBase} from "./PoolTestBase.sol";
+import {Constants} from "../../test/utils/Constants.sol";
+import {Test} from "forge-std/Test.sol";
+
+contract SkipCallsTestHook is BaseTestHooks, Test {
+    using PoolIdLibrary for PoolKey;
+    using Hooks for IHooks;
+
+    uint256 public counter;
+    IPoolManager manager;
+
+    function setManager(IPoolManager _manager) external {
+        manager = _manager;
+    }
+
+    function beforeInitialize(address, PoolKey calldata key, uint160 sqrtPriceX96, bytes calldata hookData)
+        external
+        override
+        returns (bytes4)
+    {
+        counter++;
+        _initialize(key, sqrtPriceX96, hookData);
+        return IHooks.beforeInitialize.selector;
+    }
+
+    function afterInitialize(address, PoolKey calldata key, uint160 sqrtPriceX96, int24, bytes calldata hookData)
+        external
+        override
+        returns (bytes4)
+    {
+        counter++;
+        _initialize(key, sqrtPriceX96, hookData);
+        return IHooks.afterInitialize.selector;
+    }
+
+    function beforeAddLiquidity(
+        address,
+        PoolKey calldata key,
+        IPoolManager.ModifyLiquidityParams calldata,
+        bytes calldata hookData
+    ) external override returns (bytes4) {
+        counter++;
+        _initialize(key, Constants.SQRT_RATIO_1_1, hookData);
```

Comment:
Maybe overkill but it seems like we are just calling the same function on the pool again (except for this.. this one does initialize).... Maybe instead we actually fuzz the function thats called inside? That way we are testing that no matter which hook callback we are in, any action on the pool called from the hook skips the call? So it could be any of initialize/swap/modifyPositon/donate